### PR TITLE
Use the Joomla way for bootstrap modals

### PR DESCRIPTION
**Pull request by dgrammatiko**
#### Exchange some class names

Most of the PRs that convert the mootools modals to bootstrap use `JHtmlBootstrap::renderModal(etc)`
but they should actually use `JHtml::_('bootstrap.renderModal', etc)`.
And here is how I found out:
Me: JHtmlBootstrap::renderModal( vs JHtml::_('bootstrap.renderModal'
The first one on my IDE with cmd + click on JHtmlBootstrap takes me to the correct file/class but the other takes me to JHtml and cmd + click on 'bootstrap.renderModal' isn’t really helpful as well. So I am a little bit confused here…
@mbabker: JHtml has some code that can let you override the various methods if called via JHtml::_. Direct calling of JHtml (and its subclasses) methods break that functionality. The downside is just as you pointed out, IDE autocompletion is out the window.

So in order to prevent any future conflicts with this code is better to change it ASAP!

Because this is only a rename and some CS work I would like to ask the maintainers to just review and commit. @wilsonge @roland-d ? 
To save other people and not to spent their time here, I provide a proof that all the relative modals still work as before. So:
administrator/components/com_contact/models/fields/modal/contact.php
screen shot 2015-05-20 at 3 46 56

administrator/components/com_content/models/fields/modal/article.php
screen shot 2015-05-20 at 3 48 26

administrator/components/com_menus/models/fields/menutype.php
screen shot 2015-05-20 at 3 49 12

administrator/components/com_menus/views/item/tmpl/edit_modules.php
screen shot 2015-05-20 at 3 50 24

administrator/templates/hathor/html/com_menus/menus/default.php
screen shot 2015-05-20 at 3 52 04

administrator/components/com_newsfeeds/models/fields/modal/newsfeed.php
screen shot 2015-05-20 at 3 53 37

administrator/components/com_templates/helpers/html/templates.php
screen shot 2015-05-20 at 3 55 10

administrator/components/com_users/helpers/html/users.php
screen shot 2015-05-20 at 3 57 04

administrator/templates/hathor/html/com_menus/menus/default.php
screen shot 2015-05-20 at 3 59 04

### CODE REVIEW

**Comment by mbabker** [comment]
This is definitely a separate CS PR, but mentioning it here because I looked at the diff.

When you have a method call with arguments on multiple lines, you should have one argument per line and the arguments are indented one tab from the opening call.  So, something like this:

```php
$html[] = JHtml::_(
	'bootstrap.renderModal',
	'menuTypeModal',
	array(
		'url' => $link,
		'title' => JText::_('COM_MENUS_ITEM_FIELD_TYPE_LABEL'),
		'width' => '800px',
		'height' => '300px',
		'footer' => '<button class="btn" data-dismiss="modal" aria-hidden="true">' . JText::_("JLIB_HTML_BEHAVIOR_CLOSE") . '</button>'
	)
);
```

File: administrator/components/com_menus/views/item/tmpl/edit_modules.php [hunk 1]
```diff
@@ -99,7 +99,8 @@
 						</span>
 					<?php endif; ?>
 				</td>
-			<?php echo JHtmlBootstrap::renderModal(
+			<?php echo JHtml::_(
+							'bootstrap.renderModal',
```
Comment (wilsonge) [inline administrator/components/com_menus/views/item/tmpl/edit_modules.php]: you only need 1 tab indent from the original - not 4 :P Same in other places in this PR

**Comment by smz** [comment]
@dgt41 Dimitris, I think you should check here too if  the `width` and `height` options are needed: here **some** modals are indeed iframes (and thus they obey to those options), but I'm anyway unsure if it is a good idea (think of high-res displays, like retina displays: what is the behavior with those? I don't have one to check...)

**Comment by dgt41** [comment]
@smz The only one that is not iframe is the ones in com_templates (the preview image), but lets do that in another PR, this one basically is CS and a class rename. I already did it for the batch @ #7003 but, if you want to help you can always raise a PR

**Comment by smz** [comment]
@dgt41 OK: Let's not mix too much stuff together. We can wait for this and all your other modal-related are merged and then make an assessment if any needs more "*fine tuning*"...

**Comment by wilsonge** [comment]
Fix the conflicts please :)

**Comment by dgt41** [comment]
@wilsonge conflicts solved!

**Comment by n9iels** [comment]
Not sure if this is the right place.
When I add a new menu item, and click on the "Select menu type" button I got the following error:
Notice: Undefined variable: tispanle in C:\MAMP\htdocs\joomla-cms\administrator\components\com_menus\views\menutypes\view.html.php on line 108

Little bug? Or a problem in my local installation?

**Comment by dgt41** [comment]
@n9iels that should be `$title`. Let me check how this got there…

**Comment by dgt41** [comment]
It’s #6467

**Comment by ghazal** [comment]
If I apply patch #6995 via Joomla! Patch Tester, I get a Fatal Error in :
_administrator/components/com_menus/models/fields/menutype.php_
Looks like it is a typo on line  89/90, a comma forgotten.
Corrected:
``` php
		$html[] = JHtml::_(
			'bootstrap.renderModal',
			'menuTypeModal',
			array(
```

**Comment by brianteeman** [comment]
@ghazal I just used the patchtester with this PR and the latest version of joomla staging from github with no problem

**Comment by dgt41** [comment]
@ghazal thanks

**Comment by ghazal** [comment]
I just wanted to be sure its been corrected. Thanks all for your quick reply.

**Comment by roland-d** [comment]
@brianteeman and @ghazal Did you test this patch successfully?
@n9iels Could you test it again?

**Comment by ghazal** [comment]
Looks like everything works fine.

**Comment by n9iels** [comment]
@test looks good to me!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6995">issues.joomla.org/joomla-cms/6995</a>.</sub>

**Comment by wilsonge** [comment]
As this is effectively code style anyhow but also as there are 2 good tests I'm merging this :) Thanks!

**Pull request closed without merging**